### PR TITLE
Trigger mediaType event from provider to notify model of audio-only HLS streams

### PR DESCRIPTION
**Pull request by egreaves**
The `adaptiveEventAudio` event notifies the flash provider that an HLS stream is audio-only. For audio only streams, the preview image disappears briefly while buffering before playback begins.  

In the html5 provider, `_videotag.videoTracks.length` will be 0 when HLS streams are audio-only. The `mediaType` event notifies the model before the provider's `load` function is called, so it's no longer necessary to call `sendMediaType` on load for HLS streams.

JW7-1914

### CODE REVIEW

File: src/js/providers/flash.js [hunk 1]
```diff
@@ -338,6 +338,10 @@ define([
                         this.trigger(events.JWPLAYER_MEDIA_ERROR, event);
                     }, this);
 
+                    _swf.on('adaptiveEventAudio', function() {
+                        this.trigger('mediaType', {mediaType: 'audio'});
+                    }, this);
```
Comment (robwalch) [inline src/js/providers/flash.js]: If you send this, then I also expect you to send `{mediaType: 'video'}`. Every provider in Flash needs to be updated to output video or audio when a new item is played. Otherwise, in mixed playlist, after an audio only stream, 'mediaType' will be stuck on 'audio' and never update.

File: src/js/providers/html5.js [hunk 2]
```diff
@@ -575,7 +579,10 @@ define([
             }
 
             _setLevels(item.sources);
-            this.sendMediaType(item.sources);
+
+            if(!item.sources.length || item.sources[0].type !== 'hls') {
```
Comment (robwalch) [inline src/js/providers/html5.js]: Why `!item.sources.length`? `sendMediaType` requires an array containing at least one item:
```
sendMediaType : function(levels) {
            var type = levels[0].type;
```

Comment (robwalch) [inline src/js/providers/html5.js]: I would expect this to say

```
if(item.sources.length && item.sources[0].type !== 'hls') {
```

File: src/js/providers/html5.js [hunk 3]
```diff
@@ -1064,6 +1071,8 @@ define([
 
         this.getSubtitlesTrack = _getSubtitlesTrack;
 
+        this.getMediaType = _getMediaType;
```
Comment (robwalch) [inline src/js/providers/html5.js]: Why was this added? Don't add methods that are not used.

File: src/js/providers/html5.js [hunk 4]
```diff
@@ -1122,6 +1131,15 @@ define([
         function _getSubtitlesTrack() {
             return _currentTextTrackIndex;
         }
+
+        function _setMediaType(videoTracks) {
+            _mediaType = videoTracks && videoTracks.length ? 'video' : 'audio';
```
Comment (robwalch) [inline src/js/providers/html5.js]: `mediaType` can be a local variable.

File: src/js/providers/flash.js (hunk 1 above)
Comment (egreaves) [inline src/js/providers/flash.js]: `this.sendMediaType(item.sources);` is called in both providers' `load` function. This occurs before the `adaptiveEventAudio` event and always sends `{mediaType: 'video'}` for HLS streams.

File: src/js/providers/flash.js [hunk 5]
```diff
@@ -338,6 +341,10 @@ define([
                         this.trigger(events.JWPLAYER_MEDIA_ERROR, event);
                     }, this);
 
+                    _swf.on('mediaType', function(e) {
+                        this.trigger('mediaType', {mediaType: e.mediaType});
```
Comment (robwalch) [inline src/js/providers/flash.js]: You should be able to just forward the event object `e` here, instead of creating a new object and copying the property we're looking for off it.
